lib/tier-utils.js

```diff
@@ -11,39 +11,184 @@ export const DEFAULT_PRESETS = [500, 1000, 2000, 5000];
+ *     SELECT DISTINCT "to"
+ *     FROM "CurrencyExchangeRates"
+ *     WHERE "from" = 'USD'
+ * ) SELECT uc."to", (
+ *     -- Keep only first digit from number (e.g. 4 -> 4, 89 -> 80, 1337 -> 1000)
+ *     SELECT LEFT(rate::TEXT, 1)::integer * POW(10, LENGTH(CAST(ROUND(rate) AS TEXT)) - 1)
+ *     FROM "CurrencyExchangeRates" e WHERE e.from = 'USD' AND e.to = uc.to ORDER BY e."createdAt" DESC LIMIT 1
+ *   ) AS rate
+ * FROM uniq_currencies uc
+ * ORDER BY rate DESC
+ * ```
+ */
+const CURRENCY_ADJUSTMENT_RATES = {
+  STD: 20000,
+  VND: 20000,
+  LAK: 10000,
+  IDR: 10000,
+  SLL: 10000,
+  UZS: 10000,
+  GNF: 8000,
+  PYG: 6000,
+  KHR: 4000,
+  MGA: 4000,
+  UGX: 3000,
+  COP: 3000,
+  MNT: 3000,
+  TZS: 2000,
+  CDF: 2000,
+  BIF: 2000,
+  MWK: 1000,
+  KRW: 1000,
+  LBP: 1000,
+  MMK: 1000,
+  RWF: 1000,
+  CLP: 800,
+  XAF: 600,
+  XOF: 600,
+  CRC: 600,
+  SOS: 500,
+  KZT: 400,
+  KMF: 400,
+  AMD: 400,
+  NGN: 400,
+  AOA: 400,
+  HUF: 300,
+  LKR: 300,
+  MRO: 300,
+  GYD: 200,
+  PKR: 200,
+  YER: 200,
+  DJF: 100,
+  JPY: 100,
+  RSD: 100,
+  KES: 100,
+  XPF: 100,
+  NPR: 100,
+  ALL: 100,
+  HTG: 100,
+  VUV: 100,
+  ISK: 100,
+  JMD: 100,
+  LRD: 100,
+  CVE: 100,
+  DZD: 100,
+  ARS: 100,
+  BDT: 90,
+  SEK: 90,
+  MAD: 90,
+  AFN: 80,
+  KGS: 70,
+  INR: 70,
+  RUB: 60,
+  MZN: 60,
+  PHP: 50,
+  GMD: 50,
+  DOP: 50,
+  MKD: 50,
+  ETB: 50,
+  UYU: 40,
+  MUR: 40,
+  NIO: 30,
+  THB: 30,
+  HNL: 20,
+  TWD: 20,
+  CZK: 20,
+  UAH: 20,
+  SRD: 20,
+  LSL: 10,
+  ZAR: 10,
+  SZL: 10,
+  MVR: 10,
+  ZMW: 10,
+  NAD: 10,
+  EGP: 10,
+  MDL: 10,
+  TJS: 10,
+  MXN: 10,
+  SCR: 10,
+  TRY: 10,
+  BWP: 10,
+  NOK: 9,
+  MOP: 8,
+  SBD: 8,
+  HKD: 7,
+  HRK: 7,
+  GTQ: 7,
+  TTD: 6,
+  CNY: 6,
+  BOB: 6,
+  DKK: 6,
+  RON: 4,
+  BRL: 4,
+  MYR: 4,
+  PLN: 4,
+  PEN: 3,
+  PGK: 3,
+  SAR: 3,
+  BYN: 3,
+  AED: 3,
+  ILS: 3,
+  QAR: 3,
+  BBD: 2,
+  TOP: 2,
+  FJD: 2,
+  GEL: 2,
+  BZD: 2,
+  XCD: 2,
+  WST: 2,
+};
+
 /**
  * Get the min authorized amount for order, in cents.
  * ⚠️ Only work with data from GQLV2.
  */
-export const getTierMinAmount = tier => {
+export const getTierMinAmount = (tier, currency) => {
+  const rate = CURRENCY_ADJUSTMENT_RATES[currency] || 1;
   if (!tier) {
     // When making a donation, min amount is $1
-    return DEFAULT_MINIMUM_AMOUNT;
+    return DEFAULT_MINIMUM_AMOUNT * rate;
   } else if (tier.amountType === AmountTypes.FIXED) {
     return tier.amount?.valueInCents || 0;
   } else if (tier.minimumAmount.valueInCents !== null) {
     return tier.minimumAmount.valueInCents;
   } else if (tier.presets?.length && min(tier.presets) === 0) {
     return 0;
   } else {
-    return DEFAULT_MINIMUM_AMOUNT;
+    return DEFAULT_MINIMUM_AMOUNT * rate;
+  }
+};
+
+const adaptPresetsForCurrency = (presets, currency) => {
+  const rate = CURRENCY_ADJUSTMENT_RATES[currency];
+  if (!rate) {
+    return presets;
+  } else {
+    return presets.map(amount => Math.round(amount * rate));
   }
```

Comment:
```suggestion
  return !rate ? presets : presets.map(amount => Math.round(amount * rate));
```

Reply:
I'm often allergic to ternary conditions, but I agree it simplifies the code here